app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -21,6 +21,20 @@ def fetch_middleware_servers
             server = persister.middleware_servers.find_or_build(eap.path)
             parse_middleware_server(eap, server)
 
+            # if immutable or in container flag is not set, check also the agent resource
```

Comment:
a side note: it's probably the best to do the refactoring right before the branch for the release on MiQ is forked, because after that the back-porting becomes more difficult